src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/AccountPicker.cs

```diff
@@ -36,37 +38,173 @@ internal class AccountPicker : IAccountPicker
         {
             _parentHandle = parentHandle;
             _logger = logger;
-            _synchronizationContext = synchronizationContext ?? throw new ArgumentNullException(nameof(synchronizationContext));
+            _synchronizationContext = synchronizationContext;
             _authority = authority;
             _isMsaPassthrough = isMsaPassthrough;
         }
 
         public async Task<WebAccountProvider> DetermineAccountInteractivelyAsync()
         {
-            WebAccountProvider result = null;
+#if WINDOWS_APP
+            await ShowPicker_UWPAsync().ConfigureAwait(true);                    
+#else
+            await ShowPicker_Win32Async().ConfigureAwait(false);
+#endif
+            return _provider;
+        }
 
-            // go back to the ui thread
-            await _synchronizationContext;
+#if !WINDOWS_APP
+        private async Task ShowPicker_Win32Async()
+        {
+            // if there is a sync context, move to it (go to ui thread)
+            if (_synchronizationContext != null)
+            {
+                await _synchronizationContext;
+            }
+
+            if (UseSplashScreen())
+            {
+                await ShowPickerWithSplashScreenAsync().ConfigureAwait(false);
+            }
+            else
+            {
+                if (_synchronizationContext == null)
+                {
+                    throw new MsalClientException(
+                       MsalError.WamUiThread,
+                       "AcquireTokenInteractive with broker must be called from the UI thread when using the Windows broker." +
+                        WamBroker.ErrorMessageSuffix);
+                }
+
+                await ShowPickerForWin32WindowAsync(_parentHandle).ConfigureAwait(false);
+
+            }
+        }
+
+        /// <summary>
+        /// Account Picker APIs do not work well with console apps because the console
+        /// window belongs to a different process, which causes a security exception. 
+        /// In general, if the parent window handle does not belong to the current process, 
+        /// we need to take control over the window handle by injecting a splash screen.
+        /// </summary>
+        private bool UseSplashScreen()
+        {
+            WindowsNativeMethods.GetWindowThreadProcessId(_parentHandle, out uint windowProcessId);
+            uint appProcessId = WindowsNativeMethods.GetCurrentProcessId();
+
+            return appProcessId != windowProcessId;
+        }
+
+        /// <summary>
+        /// The account picker API has bug that prevent correct usage from console apps. 
+        /// To workaround, show a splash screen and attach to it.
+        /// </summary>
+        /// <returns></returns>
+        private Task<bool> ShowPickerWithSplashScreenAsync()
+        {
+
+            if (Thread.CurrentThread.GetApartmentState() == ApartmentState.MTA)
```

Comment:
Console apps do not have UI thread, current thread will always be MTA for console apps. This will cause the Account Picker to crash. To fix this, we will create a STA thread (i.e. UI thread). 

Note that a WinForms / WPF app can have multiple STA threads, but invoking the Account Picker from an STA thread other than the UI thread will cause a crash. 

So... to sum up

- console app - create an STA thread
- non-console apps - move to the existing STA thread (or throw a meaningful exception if AcquireTokenInteractice is not called from UI thread).